src/test/java/org/opensearch/sdk/TestExtensionsRunner.java

```diff
@@ -81,7 +81,7 @@ public class TestExtensionsRunner extends OpenSearchTestCase {
     @BeforeEach
     public void setUp() throws Exception {
         super.setUp();
-        this.extensionsRunner = new ExtensionsRunner();
+        this.extensionsRunner = new ExtensionsRunnerForTest();
```

Comment:
At this point the test blocks
On digging in `ExtensionsRunner`, it tries to start `ActionListener` with timeout 0 which further has a `Thread.join()` that blocks the calling thread. 
https://github.com/opensearch-project/opensearch-sdk-java/blob/8fbcc6eaf52e6b00f8bc93e3f4256ccf0d8e794e/src/main/java/org/opensearch/sdk/ExtensionsRunner.java#L156

https://github.com/opensearch-project/opensearch-sdk-java/blob/8fbcc6eaf52e6b00f8bc93e3f4256ccf0d8e794e/src/main/java/org/opensearch/sdk/ActionListener.java#L79

Reply:
> At this point the test blocks

Yeah, that's the one that just starts it and leaves it running for testing/debug purposes.  Put a nonzero timeout in there for tests.

Reply:
Also the current `build.gradle` "run" will direct to the main method you just deleted.  Perhaps that "ExtensionsRunnerForTesting" should have a main method to be runnable there...

Reply:
If I put a `Main` method in `ExtensionsRunnerForTests` we can't set it as `mainClassName` in gradle because it is part of test sources.

Reply:
Opened #175 to address this.